src/libAtomVM/term.h

```diff
@@ -306,40 +322,22 @@ static inline bool term_is_boxed(term t)
     return ((t & 0x3) == 0x2);
 }
 
-/**
- * @brief Checks if a term is a movable boxed value
- *
- * @details Returns \c true if a term is a boxed value that can be safely copied with memcpy.
- * @param t the term that will checked.
- * @return \c true if check succeeds, \c false otherwise.
- */
-static inline bool term_is_movable_boxed(term t)
-{
-    /* boxed: 10 */
-    if ((t & 0x3) == 0x2) {
-        const term *boxed_value = term_to_const_term_ptr(t);
-        switch (boxed_value[0] & TERM_BOXED_TAG_MASK) {
-            case 0x10:
-                return true;
-
-            default:
-                return false;
-        }
-    } else {
-        return false;
-    }
-}
-
 /**
  * @brief Returns size of a boxed term from its header
  *
- * @details Returns the size that is stored in boxed term header most significant bits.
+ * @details Returns the size that is stored in boxed term header most significant bits for variable size boxed terms.
  * @param header the boxed term header.
  * @return the size of the boxed term that follows the header. 0 is returned if the boxed term is just the header.
  */
 static inline size_t term_get_size_from_boxed_header(term header)
 {
-    return header >> 6;
+    int masked_value = header & TERM_BOXED_TAG_MASK;
+    switch (masked_value) {
```

Comment:
I wasn't very happy with the size being encoded in boxed items that have a fixed size, and used all these bits for external things to encode the node name (an atom index).
Should we really encode the size here?

Reply:
`term_get_size_from_boxed_header` (and hence `term_boxed_size`) are called in 35 different places, some of them are hot paths (like GC stuff).
Without this change those functions are replaced at low level with just a bit shift, but this change will require introducing in all those 35 places an additional conditional jump.
I also don't expect external pids being the most frequent type around (compared to tuples or binaries), so this optimization have a minor impact on saved memory.

I think an optimization for fixed size boxed binaries might be doable, but right now feels a bit premature and beyond the scope (if we do that in a more extensive way it might introduce a lot of noise in this context).

Reply:
OK. I factorized some unnecessary calls of the function in memory.c.